pkg/sdn/plugin/endpoints.go

```diff
@@ -0,0 +1,190 @@
+package plugin
+
+import (
+	"fmt"
+	"net"
+	"os/exec"
+	"strings"
+
+	log "github.com/golang/glog"
+
+	kapi "k8s.io/kubernetes/pkg/api"
+	utilruntime "k8s.io/kubernetes/pkg/util/runtime"
+	//	"k8s.io/kubernetes/pkg/util/sets"
+	utilwait "k8s.io/kubernetes/pkg/util/wait"
+	"k8s.io/kubernetes/pkg/watch"
+)
+
+const (
+	monitorCmd         = "monitor"
+	unmonitorCmd       = "unmonitor"
+	clearMonitoringCmd = "clearmonitoring"
+)
+
+type endpointInfo struct {
+	ip       string
+	port     int32
+	protocol kapi.Protocol
+}
+
+func (oc *OsdnNode) EndpointStartNode() error {
+	go utilwait.Forever(oc.watchEndpoints, 0)
```

Comment:
I'll coordinate with @dcbw to make sure we don't.  But the CNI work is not landing until 3.4.